### PR TITLE
Add lang spec rules

### PR DESCRIPTION
**Pull request by zupon**
This PR adds some language-specific rewrite rules in accordance with our normalization experiments, as well as a more language-general function that can apply to a few of our chosen languages. The additions include:
- Afrikaans rule to turn contractions `'k` and `'t` into `ek` and `het`.
- Amharic rule to reduce over-differentiation of graphemes, combining two emphatic `s` graphemes, two glottal stop graphemes, and four `h` graphemes.
- Hausa rules to convert to y-with-right-hook from apostrope+y or vice versa.
- Igbo rules to convert to or from the New Standard Alphabet.
- Malagasy rule to convert standalone abbreviation `@` to full form `amin'ny`.
- Static method to strip all unicode diacritics by decomposing the string and filtering out nonspacing mark `Mn` characters

These additions, in addition to our existing rules for Zulu and Malagasy, will cover all of our languages except Somali and Swahili. Bambara, Wolof, and Yoruba are only covered by the more general strip-diacritics method and don't have any other language-specific rules at this time.

### CODE REVIEW

File: config/af.py [hunk 1]
```diff
@@ -14,6 +14,14 @@
 
 NUMERALS = byte.DIGIT
 
+EMBIGGEN_CONTRACTIONS = cdrewrite(
```
Comment (crewee) [inline config/af.py:20]: Haha, love the Simpsons reference

File: config/am.py [hunk 2]
```diff
@@ -53,6 +53,68 @@
 NUMERALS = union(byte.DIGIT,
                  utils.GEEZ_NUMERALS)
 
+REDUCE_H = union(cross("ሐ", "ሀ"),
```
Comment (crewee) [inline config/am.py]: Can we use "string_map" here?

File: config/ha.py [hunk 3]
```diff
@@ -13,6 +13,22 @@
...
+    "",
+    byte.BYTES.closure())
+
+HOOK_TO_APOSTROPHE_Y = cdrewrite(
+    cross("ƴ", "ʼy"),
+    "",
+    "",
+    byte.BYTES.closure())
+
+#LANGUAGE_SPECIFIC_PREPROCESSING = APOSTROPHE_TO_HOOK_Y
```
Comment (crewee) [inline config/ha.py:32]: Are we going to toggle these on and off?  You'll probably have to comment out the unused rewrite too if pylint complains that it's unused.

File: config/ig.py [hunk 4]
```diff
@@ -7,14 +7,28 @@
...
-                  "ị", "ň", "ñ", "ṅ", "ọ", "ụ")
+                  "ị", "ň", "ñ", "ṅ", "ọ", "ụ", "ö")
 
 INITIAL_PUNCTUATION = utils.DEFAULT_INITIAL_PUNCTUATION
 
 FINAL_PUNCTUATION = utils.DEFAULT_FINAL_PUNCTUATION
 
 NUMERALS = byte.DIGIT
 
+TO_NEW_STANDARD_ALPHABET = cdrewrite(
```
Comment (crewee) [inline config/ig.py:21]: Could you add some comments to explain the rewrites?

File: config/mg.py [hunk 5]
```diff
@@ -20,7 +20,14 @@
...
     byte.BYTES.closure())
 
-LANGUAGE_SPECIFIC_PREPROCESSING = MG_VELAR_NASAL
+MG_ABBREVIATION = cdrewrite(
+    cross("@", "amin'ny"),
+    union("[BOS]", byte.SPACE),
+    union("[EOS]", byte.SPACE),
+    byte.BYTES.closure())
+
+#LANGUAGE_SPECIFIC_PREPROCESSING = MG_VELAR_NASAL
```
Comment (crewee) [inline config/mg.py]: remove this comment?

**Review by crewee — CHANGES_REQUESTED** [review]

File: config/mg.py (hunk 5 above)
Comment (zupon) [reply config/mg.py]: Malagasy currently has two language-specific rules (the velar nasal one and the abbreviation one). Do we want to experiment with each of them individually, or will we want to combine them and do one experiment for normalizing Malagasy? If the latter, we can delete the comment but change the next line to 
`LANGUAGE_SPECIFIC_PREPROCESSING = MG_VELAR_NASAL @ MG_ABBREVIATION`.

Comment (crewee) [reply config/mg.py]: Oh I see, yeah, we can definitely experiment with adding both fixes at the same time :).

File: config/ig.py (hunk 4 above)
Comment (zupon) [reply config/ig.py:21]: Added comments to all the language-specific rules!

File: config/af.py (hunk 1 above)
Comment (zupon) [reply config/af.py:20]: It's a perfectly cromulent way to describe what this rule does

File: config/am.py (hunk 2 above)
Comment (zupon) [reply config/am.py]: Switched to string_map across the board! It did turn out that I was just missing some of the various parentheses

File: config/mg.py (hunk 5 above)
Comment (zupon) [reply config/mg.py]: Done!

File: config/ha.py (hunk 3 above)
Comment (zupon) [reply config/ha.py:32]: Pylint doesn't appear to complain here, but I'll comment the unused rewrite out for now just to be safe, and just switch them out when I try the other experiment.

File: config/am.py [hunk 6]
```diff
@@ -53,6 +53,73 @@
 NUMERALS = union(byte.DIGIT,
                  utils.GEEZ_NUMERALS)
 
+# Amharic "over-differentiates" H graphemes, emphatic S graphemes, and glottal
+# stop graphemes, which were all inherited from Ge'ez. Surveys suggest that
+# Amharic speakers prefer one form over the others. These rules convert the
+# dispreferred series graphemes to the one preferred series, when available.
```
Comment (crewee) [inline config/am.py:59]: Just one more nit, could you add a link to the paper in the comment?  Thanks!

File: config/ha.py (hunk 3 above)
Comment (crewee) [reply config/ha.py:32]: SGTM!

**Review by crewee — APPROVED** [review]
LGTM after last nit :)

File: config/am.py (hunk 6 above)
Comment (zupon) [reply config/am.py:59]: Done!

**Pull request merged**